scripts.py

```diff
@@ -391,6 +381,7 @@ def do_run(self):
 
                 self.log.debug("Completed %s", provider.service_name)
                 providers.remove(provider)
+        return progress
```

Comment:
I wrote a test that needed to see what happened inside `do_run`, and since it can potentially do multiple things it made the most sense to return a list of CoverageProviderProgress objects.